compress.c

```diff
@@ -597,30 +596,27 @@ void comprexx(char *fileptr)
 
 						if (inbuf[0] != 'y') {
 							fprintf(stderr, "%s not overwritten\n", ofname);
-							exit_code = 1;
-							return;
+							goto error;
 						}
 					}
 
 					if (unlink(ofname)) {
 						fprintf(stderr, "Can't remove old output file\n");
 						perror(ofname);
-						exit_code = 1;
-						return;
+						goto error;
 					}
 				}
 
 				if ((fdout = open(ofname, O_WRONLY|O_CREAT|O_EXCL|O_BINARY, 0600)) == -1) {
 					perror(tempname);
-					return;
```

Comment:
Note that here was not set `exit_code = 1`, while it IMO should be.

Reply:
that sounds fine